docs/tutorials/semantic_search/asymmetric_embedding_model.md

```diff
@@ -0,0 +1,532 @@
+# Tutorial: Running Asymmetric Semnantic Search within OpenSearch
+
+This tutorial demonstrates how generating text embeddings using an asymmetric embedding model in OpenSearch. The example model used in this tutorial is the multilingual
+`intfloat/multilingual-e5-small` model from Hugging Face.
+In this tutorial, you'll learn how to prepare the model, register it in OpenSearch, and run inference to generate embeddings.
+
+> **Note**: Make sure to replace all placeholders (e.g., `your_`) with your specific values.
+
+---
+
+## Prerequisites
+
+- OpenSearch installed on your machine
+- Access to the Hugging Face `intfloat/multilingual-e5-small` model (or another model of your choice).
+- Basic knowledge of Linux commands
+---
+
+## Step 1: Start OpenSearch
+
+See here for directions to install and run [OpenSearch](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/index/). 
+
+Run OpenSearch and ensure the following steps are completed. In this example, we set up the OS cluster locally.
+
+###  Update cluster settings
+
+Ensure your cluster is configured to allow registering models. You can do this by updating the cluster settings using the following request:
+
+```
+PUT _cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.allow_registering_model_via_url": "true",
+    "plugins.ml_commons.only_run_on_ml_node": "false",
+    "plugins.ml_commons.model_access_control_enabled": "true",
+    "plugins.ml_commons.native_memory_threshold": "99"
+  }
+}
+```
+
+This configuration ensures that OpenSearch can accept machine learning models from external URLs and can run models across non-ML nodes.
+
+
+---
+
+## Step 2: Prepare the model for OpenSearch
+
+In this tutorial, you’ll use the Hugging Face `intfloat/multilingual-e5-small` model, which is capable of generating multilingual embeddings. Follow these steps to prepare and zip the model for use in OpenSearch.
+
+### a. Clone the model from Hugging Face
+
+To download the model, use the following steps:
+
+1. Install Git Large File Storage (LFS) if you haven’t already:
+
+   ```
+   git lfs install
+   ```
+
+2. Clone the model repository:
```

Comment:
Rather than lfs install or cloning the whole repository may be we can do?


Create a Directory for the Files:

```
mkdir multilingual-e5-model
cd multilingual-e5-model
```

Download the Files Using wget or curl: Use the raw file links from the Hugging Face repository.


```
wget https://huggingface.co/intfloat/multilingual-e5-small/resolve/main/onnx/model.onnx
wget https://huggingface.co/intfloat/multilingual-e5-small/resolve/main/tokenizer.json
wget https://huggingface.co/intfloat/multilingual-e5-small/resolve/main/sentencepiece.bpe.model
```

Alternatively, use curl:

```
curl -O https://huggingface.co/intfloat/multilingual-e5-small/resolve/main/onnx/model.onnx
curl -O https://huggingface.co/intfloat/multilingual-e5-small/resolve/main/tokenizer.json
curl -O https://huggingface.co/intfloat/multilingual-e5-small/resolve/main/sentencepiece.bpe.model
```

Reply:
I hear you on this but lets suppose there is a new update on the model. user would have to delete and make a separate request. Also I think separate requests would cause issues since these are big files users are working on. Also its possible the endpoint can change or someone changes directories for a clean up. We lock ourselves into maintaining a tutorial for their changes

Reply:
>lets suppose there is a new update on the model. user would have to delete and make a separate request.

Yeah, if they want to use the new updated model, don't they need to do the same for your case to use the updated model? They need to pull the updates from the git.

>its possible the endpoint can change or someone changes directories for a clean up. We lock ourselves into maintaining a tutorial for their changes

I think this is true for both cases, if they change the file name, we need to update the doc in both cases, right?


My thought process is, in this way:

1. customer doesn't need to install git & git lfs
2. customer doesn't need to downlaod unnecessary package which they don't need anymore.


This is not a blocker, but a suggestion.